Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -399,4 +398,201 @@ To set up and use PGXN to publish and distribute your extension to other users,
+
+5.  **Publish Your Extension:** Use the PGXN command-line tool or web interface to publish your extension. Provide the distribution package created in the previous step and upload it to PGXN. The tool will validate the package and extract the metadata.
+
+### Updating an Extension on PGXN:
+#### New Directory Hierarchy:
+```
+my_extension/
+├── changes
+├── META.json
+├── README.md
+├── my_extension--1.0.0.sql
+├── my_extension--1.0.1.sql
+├── expected/
+│   └── my_extension--regress.out
+├── sql/
+│   └── my_extension--regress.sql
+├── my_extension.control
+├── results/
+│   └── my_extension--regress.out
+└── Makefile
+```
+
+#### Step 1: Update the Extension Control File:
+- Open the "my_extension.control" file and update the following content:
+
+```control
+# my_extension.control
+comment = 'Minimal Viable Product'
+default_version = '1.0.1'
+relocatable = true
+```
+
+#### Step 2: Create the new Extension SQL File:
+
+- Create a file named `my_extension--1.0.1.sql`.
+
+  - This file contains the new SQL statements for the extension.
+  - The format for a PostgreSQL extension file name is `extension_name--version.sql`, where `extension_name` is the name of the extension, and `version` is the version number. The double-dash serves as a delimiter to indicate the separation between the extension name and the version number.
+
+```bash
+$ touch my_extension--1.0.1.sql
+```
+
+- Open the `my_extension--1.0.1.sql` file and add the following content:
+  - The `complex_add` is a new function named complex_add that takes an array of integers as input. Inside the function, we use the unnest function to expand the array into individual elements and then calculate the sum using the SUM function. The result is returned as an integer.
+
+```sql
+-- Update path script for version 1.0.1
+-- Create necessary objects for version 1.0.1
+CREATE TABLE my_table (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(100) NOT NULL
+);
+
+-- Add 2 numbers using add function
+CREATE FUNCTION add(a integer, b integer) RETURNS integer
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT
+    RETURN a + b;
+
+CREATE FUNCTION complex_add(integer[]) RETURNS integer
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT
+    AS $$
+        SELECT COALESCE(SUM(val), 0)
+        FROM unnest($1) AS t(val)
+    $$
+;
+```
+### Step 3: Update the Makefile:
+
+- In the given code, the changes made to the `Makefile` are related to the updated value of the `DATA` variable.
+
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.1.sql
+REGRESS = my_extension--regress
+
+PG_CONFIG  ?= pg_config
+PGXS := $(shell $(PG_CONFIG) --pgxs)
+include $(PGXS)
+```
+
+## Implementation of the Extension:
+Once you have successfully implemented the `my_extension` in your PostgreSQL database, you can use the new funtion as follows:
```

Comment:
This is true if you install it and run `CREATE EXTENSION` where it did not exist before, but if it already exists the new function won't appear. That will require creating another file, `my_extension--1.0.0--1.0.1.sql` that contains just the new objects, including it in the `DATA` variable in the Makefile, and then, after installing, running `ALTER EXTENSION UPDATE`.

Reply:
Dropped the PGXN and added 2 ways of updating the extension:
- `ALTER EXTENSION`
- `DROP EXTENSION` and `CREATE EXTENSION`

I thinks the basic SQL quick start is pretty much done.